### ISSUE TITLE
MvM Attributes "AlwaysCrit" applies Crit boost to primary weapon if set on a melee weapon

### ISSUE THREAD

**Issue by SoulBlast35** [nosoop/SM-TFCustomAttributeStarterPack#58]
**Initial troubleshooting checklist**
- [ ] The attribute is loaded, or this bug report is about a plugin load failure.  (Check `sm plugins list` on the server.)
- [x ] The attribute is the most recent revision.

**Describe the bug**
If the "MvM Attributes" attribute is used to apply a perma-critboost to a Melee weapon, it will instead apply the boost to the Primary weapon.

**To Reproduce**
Apply "MvM Attributes" "AlwaysCrit" to any custom melee weapon

**Expected behavior**
The Melee Weapon should be critboosted (Interesting to note, the Spy's Knife, while it does not deal crits in this state, will play the Backstab animation on every swing)

**Server environment (please complete the following information):**
 - Operating System: Windows 10
 - SourceMod version: 1.10.0.6528

**Additional context**
An extension to the above quirk, because Spy, internally, has no "Primary" weapon, this bug does not cause his Revolver to become critboosted.

**Comment by nosoop** [comment]
Ah, I see what code handles this.

I don't remember if the person that requested this plugin specified if `AlwaysCrit` was supposed to only be active on a given item or applied across the board if any item had it; will review the conversation on it when I have the time.

That said, it does look like the condition being removed on switch is caused by the game, and that would be a bug in any case.

**Comment by nosoop** [comment]
Yeah, upon review I'm fairly sure `mvm attributes` is supposed to act as a passive property (the resulting flags is the combined result of the flags on all weapons).

The bug would be critical glows not rendering.

**Comment by SoulBlast35** [comment]
I have sort of been able to partially fix this myself by changing the TFCond to "CritOnKill", however, it still affects all weapons.

**Comment by nosoop** [comment]
> it still affects all weapons

That is intentional; the attribute affects the player whether or not the item with the attribute is active or not.  `addcond while active` is likely the preferred attribute for the case when a condition should only be active when a weapon is out.

I've added a note to clarify this in the wiki entry.
